testing/scenariobuilder/validations.go

```diff
@@ -12,17 +13,38 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-func validateTodoErrs(programOutput string, scenarios []*TodoErrScenario) validateFunc {
+func validateTodoErrs(programOutput string, scenarios []*TodoErrScenario, format string) validateFunc {
```

Comment:
I feel like it would be harder to maintain this function moving forward if we stick to this approach with instrumenting `if format == "json"` wherever its needed.

Instead, let's make two functions `validateStandardTodoErrs` and `validateJSONTodoErrs`.
In this function, you'd do:
```go
if format == "json" {
    return validateJSONTodoErrs(programOutput, scenarios)
} else {
    return validateStandardTodoErrs(programOutput, scenarios)
}
```

Reply:
This will enable you to validate todo errors as its most suitable to do so for each format.

Currently, when validating json TODO errors, you are trying to fit the json-specific validation with the standard validation and it's quite unnatural.

By extracting these two functions, you will be able to validate both formats in the best suitable way for each.